### DISCUSSION TITLE
How to easily check if a request's body is empty

### DISCUSSION THREAD

**Discussion by LukasKalbertodt** [hyperium/hyper#2672] [Q&A] [answered]
Hi there!

I'm not interested in the body's contents, I just want to know whether it is empty. 

- I can of course check the `Content-Length` header, but that is not required to be set correctly. 
- Then there is the `HttpBody::size_hint`, but that's also just a hint that I cannot rely upon.
- Of course I could just download the whole body with `body::to_bytes` or `body::aggregate`, but that is wasted time and is also not great for non-trusted inputs because it has no length checks.
- Then there are a couple other methods on `HttpBody`, but it's not clear to me how exactly they work and what guarantees they give (e.g. if `data()` returns `None`, does that mean the body is empty?)

So yes: how to easily check whether a request's body is empty? Thanks!

**Comment by sfackler** [comment, marked as answer]
The body is empty if data returns none.

**Reply by LukasKalbertodt** [reply]
Thanks! Would it make sense to put this in the documentation? I think the current wording is a bit ambiguous about that.

**Reply by seanmonstar** [reply]
Also, the size hint can be depended on, if set. It defaults to the widest range, but if an implementation sets an exact size, like 0, it will be exactly that.
